apis/bigquery/v1beta1/dataset_reference.go

```diff
@@ -71,10 +72,42 @@ func (r *DatasetRef) NormalizedExternal(ctx context.Context, reader client.Reade
 		return "", fmt.Errorf("reading referenced %s %s: %w", BigQueryDatasetGVK, key, err)
 	}
 	// Get external from status.externalRef. This is the most trustworthy place.
-	actualExternalRef, _, err := unstructured.NestedString(u.Object, "status", "externalRef")
+	actualExternalRef, found, err := unstructured.NestedString(u.Object, "status", "externalRef")
 	if err != nil {
 		return "", fmt.Errorf("reading status.externalRef: %w", err)
 	}
+	if !found {
+		// BigQueryDataset is still TF based so we resolve the reference from the object
+		// Fetch BigQueryDataset object to construct the external form.
+		dataset := &unstructured.Unstructured{}
```

Comment:
Didn't we just read the dataset before this block? Can we reuse it?

Reply:
ah good call!